src/ansys/fluent/core/filereader/casereader.py

```diff
@@ -0,0 +1,115 @@
+"""Reader for Fluent case files.
+
+Example
+-------
+
+>>> from ansys.fluent.parametric.local.filereader.casereader import CaseReader
+
+Instantiate a case reader
+
+>>> reader = CaseReader(case_filepath=case_filepath)
+
+Get lists of input and output parameters
+
+>>> input_parameters = reader.input_parameters()
+>>> output_parameters = reader.output_parameters()
+"""
+
+from typing import List
+
+import h5py
+
+from . import lispy
+
+
+class InputParameter:
+    """Class to represent an input parameter.
+
+    Attributes
+    ----------
+    name : str
+    value
+        The value of this input parameter, usually
+        a string, qualified by units
+    """
+
+    def __init__(self, raw_data):
+        self.name, self.name = None, None
+        for k, v in raw_data:
+            if k == "name":
+                self.name = v
+            elif k == "definition":
+                self.value = v
+
+
+class OutputParameter:
+    """Class to represent an output parameter.
+
+    Attributes
+    ----------
+    name : str
+    """
+
+    def __init__(self, raw_data):
+        parameter = raw_data[1]
+        for elem in parameter:
+            if len(elem) and elem[0] == "name":
+                self.name = elem[1][1]
+
+
+class CaseReader:
+    """Class to read a Fluent case file.
+
+    Methods
+    -------
+    input_parameters
+        Get a list of input parameter objects
+    output_parameters
+        Get a list of output parameter objects
+    """
+
+    def __init__(self, case_filepath: str):
+        file = h5py.File(case_filepath)
```

Comment:
Does this work with legacy .cas and .cas.gz files?

Reply:
@mkundu1 thanks for reminding me. I intended to put a check for that.

Reply:
...and we can also work on legacy support. Adding issue.